sdk/core/azure-core/test/ut/json_test.cpp

```diff
@@ -17,6 +20,16 @@ TEST(Json, create)
   EXPECT_EQ(expected, j.dump());
 }
 
+TEST(Json, customExceptionsDontEscape)
+{
+  json jsonRoot = json::parse(R"({"KeyName": 1, "AnotherObject": {"KeyName": 2}})");
+  Azure::Nullable<std::string> dest;
+
+  // Setting a number field to string results in a type mismatch error.
+  EXPECT_THROW(
```

Comment:
Please add a line verifying that exception's `what()` is what we expect.
If we end up with the template solution, please write a test case verifying each type (`bool`, `std::string` etc).